### PR TITLE
Refactor v13 metadata types

### PR DESCRIPTION
**Pull request by ascjones**
Replace custom definitions for errors and event metadata with scale-info type information for the pallet `Event` and `Error` types (depends on https://github.com/paritytech/scale-info/pull/87 which adds documentation to the `scale-info` derive). 

As @thiolliere [pointed out](https://github.com/paritytech/substrate/pull/8615#discussion_r613292720), metadata for events and errors (which are expanded to simple enums) can be captured already by the `scale_info` derives. 

Call definitions could also just use the `scale-info` derives on the `Call` enum also, but we will keep the custom types in case we need to add additional metadata, see the [comment below](https://github.com/paritytech/frame-metadata/pull/11#issuecomment-827556297).

Also: 

- Renames types/comments with module -> pallet
- Adds wrapper types for calls and events metadata

### CODE REVIEW

**Comment by thiolliere** [comment]
about calls I think it might makes sense to still organize it as metadata V12.

In the future we might want to add additional information per call variant. Like the origin the call can accept. Or just some unstructured information about the call. https://github.com/paritytech/substrate/issues/6753

In this context having the call declared as a vec of function would allow use to just add another field in this FunctionMetadata.
On the other end we can still have an extra field in ModuleMetadata: `calls_extra: Vec<FunctionMetadata>` which only contains the additional information.

**Comment by ascjones** [comment]
Makes sense, so for events and errors just the `TypeSpec` is okay?

**Comment by thiolliere** [comment]
> Makes sense, so for events and errors just the `TypeSpec` is okay?

yes for me it makes sense considering those types are only description. But maybe opinion from other people using the metadata can be helpful here.

File: frame-metadata/src/v13.rs [hunk 1]
```diff
@@ -122,44 +122,44 @@ impl IntoPortable for SignedExtensionMetadata {
...
 				.map(|constant| registry.map_into_portable(constant)),
-			errors: registry.map_into_portable(self.errors),
+			errors: self.errors.into_portable(registry),
 			index: self.index,
 		}
 	}
 }
 
-/// All metadata of the storage.
+/// All metadata of the storage.module
```
Comment (dvdplm) [inline frame-metadata/src/v13.rs:162]: Was this on purpose?

Comment (ascjones) [reply frame-metadata/src/v13.rs:160]: Er no :sweat_smile:

**Review by dvdplm — APPROVED** [review]

**Review by thiolliere — APPROVED** [review]
looks good to me

**Pull request merged**